cocotb/share/lib/gpi_log/gpi_logging.c

```diff
@@ -149,50 +159,82 @@ void gpi_log(const char *name, long level, const char *pathname, const char *fun
     // calling args is level, filename, lineno, msg, function
     //
     PyGILState_STATE gstate = PyGILState_Ensure();
+    PyObject *pLogArgs = NULL;
 
-    PyObject *check_args = PyTuple_New(1);
-    PyTuple_SetItem(check_args, 0, PyLong_FromLong(level));
+    PyObject *pCheckArgs = PyTuple_New(1);                  // New reference
+    if (pCheckArgs == NULL)
+        goto error;
 
-    PyObject *filter_ret = PyObject_CallObject(pLogFilter, check_args);
-    Py_DECREF(check_args);
-    if (filter_ret == NULL) {
-        PyErr_Print();
-        PyGILState_Release(gstate);
-        return;
-    }
-    int is_enabled = PyObject_IsTrue(filter_ret);
-    Py_DECREF(filter_ret);
+    PyObject *pLevel = PyLong_FromLong(level);              // New reference
+    if (pLevel == NULL)
+        goto error;
+    if (0 != PyTuple_SetItem(pCheckArgs, 0, pLevel))        // Note: This function steals a reference to pLevel
+        goto error;
```

Comment:
This is overkill - you could use `PyTuple_SETITEM` here instead, which always succeeds (and below)

Reply:
Should we check the `_From` function returns or assume everything will work as before?

Reply:
The `_From` functions _can_ fail - they can throw `MemoryError` - so your change is an improvement there, albeit an irrelevant one. See my comment below though - using `PyObject_CallFunction` or `PyBuildValue` makes this type of tuple construction far less painful.